src/pages/api/secure/send-change-email-link.ts

```diff
@@ -0,0 +1,62 @@
+import jwt from 'jsonwebtoken'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { withSessionRoute } from '@/ironAuth/withSessionApiRoute'
+import { appUrl } from '@/utils/constants'
+import { sendChangeEmailEmail } from '@/utils/email_helper'
+
+const JWT_SECRET = process.env.JWT_SECRET
+const TOKEN_EXPIRY = '5m'
+
+if (!JWT_SECRET) {
+  throw new Error('JWT_SECRET environment variable is required')
+}
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'POST') {
+    return res.status(405).json({ error: 'Method not allowed' })
+  }
+
+  try {
+    const account_address = req.session.account!.address
+
+    // Create a secure JWT with expiration and single-use capability
+    const payload = {
+      type: 'change_email',
+      account_address,
+      iat: Math.floor(Date.now() / 1000),
+      jti: `${account_address}-${Date.now()}-${Math.random()
+        .toString(36)
+        .substr(2, 9)}`,
+    }
+
+    const changeToken = jwt.sign(payload, JWT_SECRET!, {
+      algorithm: 'HS256',
+      expiresIn: TOKEN_EXPIRY,
+    })
+
+    const changeUrl = `${appUrl}/dashboard/change-email?token=${changeToken}&address=${account_address}`
+
+    // Get user's current notification email from the request body
+    const { currentEmail } = req.body
```

Comment:
The current email is already in the DB and not required on the request; otherwise, a mail can be injected while making the request. The email on the DB is immutable and should be the single source of truth.